feat: limit x-axis range

rephus opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Closes:  #2943

### Description:

This issue https://github.com/apache/echarts/issues/16984#issuecomment-1849116907 is still affecting these charts. 

![Screenshot from 2023-12-11 11-41-14](https://github.com/lightdash/lightdash/assets/1983672/fb56fe50-73d4-4112-81a2-0620dee92f60)

![Screenshot from 2023-12-11 11-41-38](https://github.com/lightdash/lightdash/assets/1983672/baf463f2-0bd8-492a-92c0-dc11c63b7310)


<!-- Add a description of the changes proposed in the pull request. -->
![Screenshot from 2023-12-11 11-41-48](https://github.com/lightdash/lightdash/assets/1983672/8b07fd9b-cf6d-4fcc-a14c-427ee2fe1cd4)

<!-- Even better add a screenshot / gif / loom -->
![Screenshot from 2023-12-11 12-03-04](https://github.com/lightdash/lightdash/assets/1983672/40fdf011-10d4-4824-849d-d1caaadc55c0)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

rephus commented:
## Known limitation

When a chart is flipped (and the xAxis becomes the yAxis) the min/max range doesn't work, 

This is because when the axis changes, it goes from `type: value` to `type: category`

This should be fixed in a separate PR 

![Screenshot from 2023-12-11 12-08-48](https://github.com/lightdash/lightdash/assets/1983672/508e58e7-abfb-4ede-ba12-4f9ca1c679da)
![Screenshot from 2023-12-11 12-09-12](https://github.com/lightdash/lightdash/assets/1983672/61a64f7f-f8c2-4493-ae5e-22da29f342e0)

ZeRego commented:
Label isn't correct. Should say `x` and `bottom` 

<img width="204" alt="Screenshot 2023-12-12 at 09 42 14" src="https://github.com/lightdash/lightdash/assets/9117144/b6b5c321-c5c0-4aaf-9454-540d15e12977">

Range input didn't show when I used order date (day)

<img width="1657" alt="Screenshot 2023-12-12 at 09 41 32" src="https://github.com/lightdash/lightdash/assets/9117144/e92112df-f14f-4f06-a57e-2468ee8bb90e">

https://lightdash-pr-8261.onrender.com/share/34jDN2xe1xKe1UP-MtHVj

ZeRego commented on packages/frontend/src/hooks/echarts/useEchartsCartesianConfig.ts:
```diff
@@ -991,14 +990,12 @@ const getEchartAxis = ({
+                    xAxisConfiguration?.[1]?.min ||
+                    maybeGetAxisDefaultMinValue(allowSecondAxisDefaultRange),
+                max:
+                    xAxisConfiguration?.[1]?.min ||
```
```suggestion
                    xAxisConfiguration?.[1]?.max ||
```

rephus replied:
good catch, thanks

rephus commented:
> Label isn't correct. Should say `x` and `bottom`

Fixed, thanks

![Screenshot from 2023-12-12 13-50-12](https://github.com/lightdash/lightdash/assets/1983672/6c63e092-ea44-4938-aba0-612c081ec9c1)

![Screenshot from 2023-12-12 13-50-03](https://github.com/lightdash/lightdash/assets/1983672/a412a89c-d7e8-45b9-9ed5-16a12ef73f91)

Also fixed dates
